packages/widget/src/components/TokenList/TokenListItem.tsx

```diff
@@ -101,7 +103,12 @@ export const TokenListItemButton: React.FC<TokenListItemButtonProps> = ({
       dense
     >
       <ListItemAvatar>
-        <Avatar src={token.logoURI} alt={token.symbol}>
+        <Avatar
+          src={token.logoURI}
+          alt={token.symbol}
+          sx={!imageLoaded ? { bgcolor: grey[400] } : null}
+          onLoad={() => setImageLoaded(true)}
```

Comment:
How does this affect performance? With this change, we will have 2x renders for every component in the list.

Reply:
Wouldn't it re-render the whole <TokenListItemButton> component when setImageLoaded is triggered? Maybe we can add a `ListItemAvatarBase` component below and move `<Avatar>` together with `imageLoaded`/`setImageLoaded` state there, so we are sure we are updating only the small `<Avatar>` component, and not the whole `<TokenListItemButton>`.

Reply:
Yeah, the entire component would be re-rendered, but I was testing with repaint flashing enabled and only the Avatar section was being repainted.